tests/ignite/contrib/handlers/test_tensorboard_logger.py

```diff
@@ -474,37 +474,52 @@ def _test(tag=None):
     _test(tag="tag")
 
 
-def test_grads_scalar_handler_frozen_layers(dummy_model_factory, norm_mock):
-    model = dummy_model_factory(with_grads=True, with_frozen_layer=True)
+def test_grads_scalar_handler_whitelist(dummy_model_factory, norm_mock):
+    model = dummy_model_factory()
 
-    wrapper = GradsScalarHandler(model, reduction=norm_mock)
+    wrapper = GradsScalarHandler(model, reduction=norm_mock, whitelist=["fc2.weight"])
     mock_logger = MagicMock(spec=TensorboardLogger)
     mock_logger.writer = MagicMock()
 
     mock_engine = MagicMock()
     mock_engine.state = State()
     mock_engine.state.epoch = 5
-    norm_mock.reset_mock()
 
+    wrapper(mock_engine, mock_logger, Events.EPOCH_STARTED)
+    mock_logger.writer.add_scalar.assert_called_once_with("grads_norm/fc2/weight", ANY, 5)
+    mock_logger.writer.reset_mock()
+
+    wrapper = GradsScalarHandler(model, tag="model", whitelist=["fc1"])
     wrapper(mock_engine, mock_logger, Events.EPOCH_STARTED)
 
     mock_logger.writer.add_scalar.assert_has_calls(
-        [call("grads_norm/fc2/weight", ANY, 5), call("grads_norm/fc2/bias", ANY, 5)], any_order=True
+        [
+            call("model/grads_norm/fc1/weight", ANY, 5),
+            call("model/grads_norm/fc1/bias", ANY, 5),
+        ],
+        any_order=True,
     )
+    assert mock_logger.writer.add_scalar.call_count == 2
+    mock_logger.writer.reset_mock()
 
-    with pytest.raises(AssertionError):
-        mock_logger.writer.add_scalar.assert_has_calls(
-            [call("grads_norm/fc1/weight", ANY, 5), call("grads_norm/fc1/bias", ANY, 5)], any_order=True
-        )
+    def weight_selector(n, _):
+        return "bias" in n
+
+    wrapper = GradsScalarHandler(model, tag="model", whitelist=weight_selector)
+    wrapper(mock_engine, mock_logger, Events.EPOCH_STARTED)
+
+    mock_logger.writer.add_scalar.assert_has_calls(
+        [
+            call("model/grads_norm/fc1/bias", ANY, 5),
+            call("model/grads_norm/fc2/bias", ANY, 5),
+        ],
+        any_order=True,
+    )
     assert mock_logger.writer.add_scalar.call_count == 2
-    assert norm_mock.call_count == 2
 
 
 def test_grads_hist_handler_wrong_setup():
 
-    with pytest.raises(TypeError, match="Argument model should be of type torch.nn.Module"):
```

Comment:
Why do you remove this check ?

Reply:
As `GradHistHandler` and its brothers all subclass `BaseWeightsHandler`, I added `test_base_weights_handler_wrong_setup` to `test_base_logger` and did this test there.